json_schema_fuzz/schema_operations.py

```diff
@@ -1,45 +1,245 @@
-                and isinstance(schema_b[key], list)
-        ):
-            # Append lists together
-            schema_a[key].extend(schema_b[key])
+    *schemas: List[Dict[Any, Any]],
+) -> Dict[Any, Any]:
+    """
+    Merge a list of JSON schemas recursively.
+
+    Merging schemas here means that all schemas
+    must be true for result schema to be true.
+    """
+
+    merged_schema = {}
+
+    # Combinations
+
+    all_of_values = get_val_or_none(schemas, "allOf")
+    if all_of_values:
+        merged_schema["allOf"] = list(
+            itertools.chain(*all_of_values)
+        )
+
+    any_of_values = get_val_or_none(schemas, "anyOf")
+    if any_of_values:
+        merged_schema["anyOf"] = []
+        for anyof_permutation in itertools.product(*any_of_values):
+            merged_schema["anyOf"].append(merge(*anyof_permutation))
+
+    one_of_values = get_val_or_none(schemas, "oneOf")
+    if one_of_values:
+
+        # Build inverse values for all schemas provided
+        inverted_oneof_values = []
+        for one_of_list in one_of_values:
+            inverted_oneof_values.append([
+                invert(s) for s in one_of_list
+            ])
+
+        new_anyof_values = []
+
+        # Build permutations where one value is true and the rest are false
+        one_of_indexes = [range(len(v)) for v in one_of_values]
+        # During this permutation, use the index provided as the one "true"
+        # and have the rest of the indexes be false
+        for true_indexes in itertools.product(*one_of_indexes):
+            # Make a copy of the inverted one_of value
+            current_permutation = copy.deepcopy(inverted_oneof_values)
+
+            # Replace inverted values with given values at the true_indexes
+            for outer_list_index, _ in enumerate(current_permutation):
+                inner_list_index = true_indexes[outer_list_index]
+                current_permutation[outer_list_index][inner_list_index] = \
+                    one_of_values[outer_list_index][inner_list_index]
+
+            # Merge and save
+            denested_schemas = [
+                subschema for outer_list in current_permutation
+                for subschema in outer_list
+            ]
+            new_anyof_values.append(merge(*denested_schemas))
+
+        # If there are already anyof values,
+        # call merge() again to merge them with the
+        # new anyOf values
+        if 'anyOf' in merged_schema:
+            merged_schema["anyOf"] = merge(
+                {"anyOf": merged_schema["anyOf"]},
+                {"anyOf": new_anyof_values}
+            )
         else:
-            raise NotImplementedError(
-                f"Conflicting key {key} encountered in path {path}")
-    if in_place:
-        return None
-    else:
-        return schema_a
+            merged_schema["anyOf"] = new_anyof_values
+
+    # Numbers
+
+    minimum_values = get_val_or_none(schemas, "minimum")
```

Comment:
We've got a bunch of stuff with this basic form:
```python
    values = get_val_or_none(schemas, keyword)
    if values:
        merged_schema[keyword] = fcn(values)
```
This could be substantially shorter, and maybe be more readable, if we explicitly generalize that operation and make a table of those `keyword, fcn` pairs.